shell/platform/embedder/embedder.h

```diff
@@ -552,6 +578,110 @@ typedef struct {
   const FlutterTaskRunnerDescription* platform_task_runner;
 } FlutterCustomTaskRunners;
 
+typedef struct {
+  FlutterOpenGLTargetType type;
+  union {
+    FlutterOpenGLTexture texture;
+    FlutterOpenGLFramebuffer framebuffer;
+  };
+} FlutterOpenGLCompositorLayerRenderer;
+
+typedef struct {
+  const void* allocation;
+  size_t row_bytes;
+  size_t height;
+  void* user_data;
+  VoidCallback destruction_callback;
+} FlutterSoftwareCompositorLayerRenderer;
+
+typedef enum {
+  kFlutterCompositorLayerRendererTypeOpenGL,
+  kFlutterCompositorLayerRendererTypeSoftware,
+} FlutterCompositorLayerRendererType;
+
+typedef struct {
+  // The size of this struct. Must be sizeof(FlutterCompositorLayerRenderer).
+  size_t struct_size;
+  // A baton that is not interpreted by the engine in any way. The embedder may
+  // use this to associate resources that are tied to the lifecycle of the
+  // |FlutterCompositorLayerRenderer|.
+  void* user_data;
+  FlutterCompositorLayerRendererType type;
+  union {
+    FlutterOpenGLCompositorLayerRenderer open_gl;
+    FlutterSoftwareCompositorLayerRenderer software;
+  };
+} FlutterCompositorLayerRenderer;
+
+typedef struct {
+  double x;
+  double y;
+} FlutterPoint;
+
+typedef struct {
+  double width;
+  double height;
+} FlutterSize;
+
+typedef struct {
+  // The size of this struct. Must be
+  // sizeof(FlutterCompositorLayerRendererConfig).
+  size_t struct_size;
+
+  // The size of the render target the engine expects to render into.
+  FlutterSize size;
+} FlutterCompositorLayerRendererConfig;
+
+typedef int64_t FlutterLayerIdentifier;
+
+typedef struct {
+  // This size of this struct. Must be sizeof(FlutterCompositorLayer).
+  size_t struct_size;
+  FlutterLayerIdentifier identifier;
+  // The offset of this layer relative to the top left of the screen.
```

Comment:
The screen, or the Flutter engine's root display area? It seems like it should be the latter.

It would also be good to specify whether this is in Flutter coordinates (I assume yes, but wouldn't hurt to be explicit). (E.g., on a macOS retina display, it would be using the 2x coordinates of the GL backing store, right?)

Reply:
It is actually not scaled. I'll add a separate discussion for this. Addressed the original comment.